server/server.go

```diff
@@ -70,9 +70,10 @@ type Server struct {
 	grpcListener net.Listener
 	httpServer   *http.Server
 
-	HTTP *mux.Router
-	GRPC *grpc.Server
-	Log  logging.Interface
+	HTTP       *mux.Router
+	HTTPServer *http.Server
```

Comment:
why do we want both `httpServer` and `HTTPServer` ?
Is the exposed one supposed to be mutable?
If not, a function would be better.

Reply:
> why do we want both httpServer and HTTPServer ?

Good point.

> Is the exposed one supposed to be mutable?

The HTTPServer isn't supposed to be mutable, no.  We just need to be able to call the (middleware wrapped) handler on it; we could just expose that instead?